### PR TITLE
Added clear inventory option

### PR DESCRIPTION
**Pull request by jellebreuer**
So I needed this feature for my own server, maybe you like it too.

I use the plugin to run a skyblock server, with this I can prevent resource sharing. With the current setup a player could create an island, harvest all the resources, go to a friends island and deposit. Then generate a new island and do the same thing. With this option, your inventory gets cleared on every command that gets you to an island. As long as you make sure there are no other plugins making this possible you should be good :)

### CODE REVIEW

**Comment by Mohron** [comment]
I'm curious about how you had your server set up. You **should** be able to prevent resource sharing by limiting all players to one island since players cannot invite a player who has reached their limit.

I can image that a feature like this would create more problems then it will solve... If I was to release it, I personnally would need to rework every command that resets a player's inventory to require confirmation to reduce the likelyhood of a player doing this by mistake. Also, with additional plugins and mods, this system could easily be bypassed. I understand your intentions but I think we should work to reach a better solution that is less harsh and not bypassable through other methods of teleportation or even just flighing to another island.

**Comment by phit** [comment]
couldnt you just cancel all inventory interaction when outside of your own claim?

**Comment by Mohron** [comment]
GP already does, unless you're trusted, which seems to be the actual issue here.

**Comment by jellebreuer** [comment]
@Mohron I see your point, only there is no way to know when you leave or enter a claim without packet inspection (correct me if I'm wrong). In my specific setup, this works great since I made sure the only way to get on an island is with "is home" "is tp" "is spawn" and " is create/reset". I understand if it won't make it in the official package. For now, I'm happy with my fork :)

**Comment by jellebreuer** [comment]
Another option I might implement is disabling the dropping/banking of any item found on a starting island. Then the clearing isn't necessary anymore.

**Comment by Mohron** [comment]
GP already makes it possible to detect players entering and exiting claims. SkyClaims already uses this to enforce the island lock feature.

**Comment by jellebreuer** [comment]
@Mohron you are right, i tried adding some code to the onClaimBorder listener in the claimeventhandler. only it doesn't seem to get triggered. I will mess around some more

**Comment by jellebreuer** [comment]
@Mohron Alright I figured it out! I updated the PR, now you can define a list of items to get removed from your inventory when you go in or out of a claim/island. It works pretty well, thanks for pushing me to make a better solution

File: src/main/java/net/mohron/skyclaims/listener/ClaimEventHandler.java [hunk 1]
```diff
@@ -112,6 +118,21 @@ public void onClaimChanged(ChangeClaimEvent event, @Root Player player,
...
   }
 
+  @Listener
+  public void onClaimBorderEvent(BorderClaimEvent e, @First Player player) {
+    for (String itemName : PLUGIN.getConfig().getMiscConfig().getClearItems()){
+      Optional<ItemType> item = Sponge.getRegistry().getType(ItemType.class, itemName);
+      if(item.isPresent()) {
+        ItemType type = item.get();
+        if (player.getInventory().contains(type)) {
+          ItemStack dirt = ItemStack.builder().itemType(type).build();
```
Comment (Mohron) [inline src/main/java/net/mohron/skyclaims/listener/ClaimEventHandler.java]: Dirt you say? :D

**Comment by jellebreuer** [comment]
@Mohron whoops xD
PS. you have to use sponge names for the config list, so "DIRT, "ICE" etc. if this gets merged I can also write some documentation for it.

**Comment by Mohron** [comment]
Merged/modified here: 7eab3e1764c52fe93903590b56fa8d254bda8545

If you still use this fuctionality, I'd appreciate if you could help test my changes once I have a build ready!

**Comment by jellebreuer** [comment]
@Mohron The server is gone and i have moved on to other games, sorry

**Pull request merged**